x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -152,6 +152,9 @@ public SemanticTextFieldMapper build(MapperBuilderContext context) {
                 throw new IllegalArgumentException(CONTENT_TYPE + " field [" + leafName() + "] does not support multi-fields");
             }
             final String fullName = context.buildFullName(leafName());
+            if (context.isNested()) {
```

Comment:
does this check work if there is a nested context, an object context that may have multiple fields? E.g. multiple layers?